lib/lambda_ethereum_consensus/utils.ex

```diff
@@ -47,4 +47,27 @@ defmodule LambdaEthereumConsensus.Utils do
     encoded = binary |> Base.encode16(case: :lower)
     "0x#{String.slice(encoded, 0, 3)}..#{String.slice(encoded, -4, 4)}"
   end
+
+  def chunk_by_sizes(enum, sizes), do: chunk_by_sizes(enum, sizes, [], 0, [])
+
+  # No more elements, there may be a leftover chunk to add.
+  def chunk_by_sizes([], _sizes, chunk, chunk_size, all_chunks) do
+    if chunk_size > 0 do
+      [Enum.reverse(chunk) | all_chunks] |> Enum.reverse()
+    else
+      Enum.reverse(all_chunks)
+    end
+  end
+
+  # No more splits will be done. We just performed a split.
+  def chunk_by_sizes(enum, [], [], 0, all_chunks), do: [enum | Enum.reverse(all_chunks)]
+
+  def chunk_by_sizes(enum, [size | rem_sizes] = sizes, chunk, chunk_size, all_chunks) do
+    if chunk_size == size do
+      chunk_by_sizes(enum, rem_sizes, [], 0, [Enum.reverse(chunk) | all_chunks])
+    else
+      [elem | rem_enum] = enum
+      chunk_by_sizes(rem_enum, sizes, [elem | chunk], chunk_size + 1, all_chunks)
+    end
+  end
```

Comment:
I may be missing some case, just tested it quickly, but i think we can simplify this to something like:
```elixir
  def chunk_by_sizes(enum, sizes) do
    Enum.reduce(sizes, {enum, []}, fn size, {remaining, acc} ->
      {chunk, rest} = Enum.split(remaining, size)
      {rest, [chunk | acc]}
    end)
    |> elem(1)
    |> Enum.reverse()
  end
```

I don't think we'll gain anything performance-wise, but its easier to read I guess.

Reply:
It's definitely easier to read! It's also similar how I implemented it originally, and it had some performance problems, which is why I did it manually. I'll merge and review it later on though! Maybe I did something wrong in the first try.